### ISSUE TITLE
Store model type and number of classes as metadata in the '.pt' file

### ISSUE THREAD

**Issue by mikel-brostrom** [Deci-AI/super-gradients#1303]
### 🚀 Feature Request

There is some manual work that has to be done to get a custom model working when it could be completely automatized.

```python
self.model = models.get(
    <type_of_yolo_model>,
    num_classes=<the-number-of-classes-in-your-model>,
    checkpoint_path =<path-to-your-weights.pt>
)
```

### Proposed Solution (Optional)

You could save the model like this:

```python
torch.save({
    'state_dict': model.state_dict(),
    'metadata': {'model_type': 'yolo_nas_s', 'classes_count': 80}
}, '/pth/to/your/model.pt')
```

Then you could load it like this::

```python
checkpoint = torch.load('model.pt')
metadata = checkpoint['metadata']

self.model = models.get(
    metadata['model_type'],
    num_classes=metadata['classes_count']>,
    checkpoint_path ='/pth/to/your/model.pt'
)
```

for loading

**Comment by 23pointsNorth** [comment]
This is a great add-on and makes it much more user-friendly. Any ideas about what's planned for the next releases?

**Comment by mikel-brostrom** [comment]
> Any ideas about what's planned for the next releases?

I don't know but this should be high on their list IMO :rofl:

**Comment by mikel-brostrom** [comment]
https://github.com/mikel-brostrom/yolo_tracking/blob/master/examples/detectors/yolonas.py
